### PR TITLE
PT-1347 | Fix validation in edit enrolment page + add tests

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Fix enrolment form validation to match with Teacher UI
- Take edited enrolment group size into account with remaining seat when validating group size

### Closes :no_good_woman:
**[PT-1347](https://helsinkisolutionoffice.atlassian.net/browse/PT-1347):** 

## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/enrolment/EditEnrolmentPage.tsx [hunk 1]
```diff
@@ -145,10 +176,18 @@ const EditorEnrolmentPage: React.FC = () => {
...
+                <EnrolmentForm
+                  onSubmit={handleSubmit}
+                  initialValues={initialValues}
+                  minGroupSize={minGroupSize}
+                  maxGroupSize={maxGroupSize}
+                />
+              ) : (
+                <Notification label="Virhe" type="error">
+                  Virhe lomakkeen alustuksessa
+                </Notification>
```
Comment (Sartonon) [inline src/domain/enrolment/EditEnrolmentPage.tsx:189]: I will ask for translatio if we are going to keep this here. Not 100% sure how we shoul to this kind of error handling, shouldn't if nothing strange happens in API

File: src/domain/enrolment/EditEnrolmentPage.tsx [hunk 2]
```diff
@@ -157,4 +196,6 @@ const EditorEnrolmentPage: React.FC = () => {
   );
 };
 
+const isNumber = (n?: number | null): n is number => typeof n === 'number';
```
Comment (nikomakela) [inline src/domain/enrolment/EditEnrolmentPage.tsx:199]: Lodash has the same, but acceptable.

File: src/domain/enrolment/EditEnrolmentPage.tsx (hunk 1 above)
Comment (nikomakela) [reply src/domain/enrolment/EditEnrolmentPage.tsx:189]: It's good to inform that something is not ok.

**Review by nikomakela — APPROVED** [review]
I tested the validation and didn't ran into any problems. There is also quite good amount of new tests related to this. Good job.

File: src/domain/enrolment/EditEnrolmentPage.tsx (hunk 2 above)
Comment (Sartonon) [reply src/domain/enrolment/EditEnrolmentPage.tsx:199]: Maybe I'll change  it

**Pull request merged**